librz/analysis/type_pdb.c

```diff
@@ -454,29 +472,30 @@ static RzType *parse_structure(const RzTypeDB *typedb, SType *type) {
  * \brief Parses union member
  *
  * \param typedb Types DB instance
- * \param type_info Current type info (member)
+ * \param stream TPI Stream
+ * \param type_info Current TpiType (member)
  * \return RzTypeUnionMember* parsed member, NULL if fail
  */
-static RzTypeUnionMember *parse_union_member(const RzTypeDB *typedb, STypeInfo *type_info) {
-	rz_return_val_if_fail(type_info && typedb, NULL);
+static RzTypeUnionMember *parse_union_member(const RzTypeDB *typedb, TpiStream *stream, TpiType *type_info) {
+	rz_return_val_if_fail(type_info && stream && typedb, NULL);
 	char *name = NULL;
 	ut64 offset = 0;
 	RzType *type = NULL;
 	switch (type_info->leaf_type) {
-	case eLF_ONEMETHOD: {
-		name = type_info->get_name(type_info);
-		type = parse_type_onemethod(typedb, type_info);
+	case LF_ONEMETHOD: {
+		name = rz_bin_pdb_get_type_name(type_info);
+		type = parse_type_onemethod(typedb, stream, type_info);
 		break;
 	}
-	case eLF_MEMBER: {
-		offset = type_info->get_val(type_info);
-		name = type_info->get_name(type_info);
-		type = parse_type_member(typedb, type_info);
+	case LF_MEMBER: {
+		offset = rz_bin_pdb_get_type_val(type_info);
+		name = rz_bin_pdb_get_type_name(type_info);
```

Comment:
do we really need to expose these apis? feels more they should private than public

Reply:
Do you mean that we should access to the field manually?

Reply:
no. i mean that normally `rz_bin_*` methods are public apis (i.e. that starts with `RZ_API <ret_type> rz_bin_*(...);`), so i think that these methods should be declare as `<ret_type> pdb_*(...);` and not exposed as public ones.

Reply:
It's two different modules.